diff --git a/g2o/core/auto_differentiation.h b/g2o/core/auto_differentiation.h
index 7d9fd5b7..2ffd93cc 100644
--- a/g2o/core/auto_differentiation.h
+++ b/g2o/core/auto_differentiation.h
@@ -158,7 +158,7 @@ class AutoDifferentiation {
   template <int EdgeDimension, int VertexDimension>
   using ADJacobianType =
       typename Eigen::Matrix<double, EdgeDimension, VertexDimension,
-                             Eigen::RowMajor>;
+                             (EdgeDimension > 1 && VertexDimension == 1) ? Eigen::ColMajor : Eigen::RowMajor>;
 
   //! helper for computing the error based on the functor in the edge
   static void computeError(Edge* that) {
